difftest/difftest.cc

```diff
@@ -638,8 +643,8 @@ void debug_mem_sync(reg_t addr, void* buf, size_t n) {
   ref->debug_memcpy_from_dut(addr, buf, n);
 }
 
-void difftest_load_flash(void *flash_bin, size_t size) {
-
+void difftest_load_flash(const uint8_t *flash_bin, size_t size) {
```

Comment:
According to NEMU (https://github.com/OpenXiangShan/NEMU/blob/master/src/cpu/difftest/ref.c#L104-L109) and DiffTest (https://github.com/OpenXiangShan/difftest/blob/master/src/test/csrc/difftest/difftest.cpp#L596), this `void *flash_bin` is a `char *` pointer to the flash file instead of a pointer to the memory contents.

I agree that an API with pointer to the flash contents is better. However, this change breaks our code in NEMU and DiffTest. Can you use this Spike version to diff with NEMU?

I would like to add a `difftest_load_flash_v2` for the new API.

Reply:
Sorry it was an oversight on my part, this interface is not used in NEMU, I'm adding implementations for both sides (NEMU and spike)